sql/src/main/java/io/crate/operation/operator/CmpOperator.java

```diff
@@ -55,4 +63,35 @@ public Boolean evaluate(Input<Object>... args) {
         }
     }
 
+    protected abstract static class CmpResolver extends BaseFunctionResolver {
+
+        private static final Param primitiveTypes = Param.of(DataTypes.PRIMITIVE_TYPES);
+
+        private final String name;
+
+
+        CmpResolver(String name) {
+            this(name, FuncParams.builder(primitiveTypes, primitiveTypes).build());
+        }
+
+        CmpResolver(String name, FuncParams funcParams) {
+            super(funcParams);
+            this.name = name;
+        }
+
+        protected abstract FunctionImplementation createInstance(FunctionInfo info);
```

Comment:
Might want to consider changing the constructor to take a `Function<FunctionInfo, FunctionImplementation>` and make this non-abstract. Would allow to use method refs (Ex. `new CmpResolver(NAME, GteOperator::new)` (and satisfy my personal composition > inheritance preference :smile: )

Reply:
Alright, whatever makes you happy 😎